### PR TITLE
Update particles.py

### PR DESCRIPTION
**Pull request by tsung1029**
Added new attribute for particles (method).  Method can be one of 4 types of orbit integrations.

Boris
Vay
Higuera
Li

### CODE REVIEW

File: PICMI_Python/particles.py [hunk 1]
```diff
@@ -19,6 +19,7 @@ class PICMI_Species(_ClassWithInit):
     Species
       - particle_type=None: A string specifying an elementary particle, atom, or other, as defined in the openPMD 2 species type extension, openPMD-standard/EXT_SpeciesType.md
       - name=None: Name of the species
+      - method: One of 'Boris', 'Vay', 'Higuera', 'Li' (string)
```
Comment (ax3l) [inline PICMI_Python/particles.py:22]: openPMD cross-link: I think for `Higuera` we do consider `Higuera-Cary` as a name, but it's still open.
https://github.com/openPMD/openPMD-standard/issues/226

Further values in openPMD:
- `free-streaming` (constantly moving with initial momentum)
- `LLRK4` (reduced Laundau-Lifshitz pusher via RK4 and classical radiation reaction, [doi:10.1016/j.cpc.2016.04.002](https://dx.doi.org/10.1016/j.cpc.2016.04.002))
https://github.com/openPMD/openPMD-standard/blob/upcoming-2.0.0/EXT_ED-PIC.md#particle-records-macroparticles
- `other:freeText` for other things a code supports

File: PICMI_Python/particles.py [hunk 2]
```diff
@@ -27,8 +28,14 @@ class PICMI_Species(_ClassWithInit):
       - particle_shape: Particle shape used for deposition and gather ; if None, the default from the `Simulation` object will be used. Possible values are 'NGP', 'linear', 'quadratic', 'cubic'
     """
 
+    methods_list = ['Boris' , 'Vay', 'Higuera', 'Li']
+    
     def __init__(self, particle_type=None, name=None, charge_state=None, charge=None, mass=None,
```
Comment (ax3l) [inline PICMI_Python/particles.py:33]: I think the `method=None,` parameter still needs to be added and then the value needs to go into a `self.method = method`?

File: PICMI_Python/particles.py (hunk 1 above)
Comment (tsung1029) [reply PICMI_Python/particles.py]: Good suggestions!  Please take a look at the new version and give feedbacks or edit.

File: PICMI_Python/particles.py (hunk 2 above)
Comment (dpgrote) [reply PICMI_Python/particles.py:33]: Yes, that is correct, method needs to be added to the parameter list.

File: PICMI_Python/particles.py [hunk 3]
```diff
@@ -27,8 +28,19 @@ class PICMI_Species(_ClassWithInit):
...
+    methods_list = ['Boris' , 'Vay', 'Higuera-Cary', 'Li', 'free-streaming', 'LLRK4']
+
     def __init__(self, particle_type=None, name=None, charge_state=None, charge=None, mass=None,
                  initial_distribution=None, particle_shape=None, density_scale=None, **kw):
+
+
+        assert method is None or method in PICMI_Species.methods_list or method.startswith('other:'), \
+            Exception('method must starts with either other:, or be one of the following '+', '.join(PICMI_Species.methods_list))    
+
+        if (method is None):
```
Comment (dpgrote) [inline PICMI_Python/particles.py:40]: The standard so far in PICMI is to leave input parameters as None if they are not specified (and not set any specific defaults). This should only be `self.method = method`.

File: PICMI_Python/particles.py [hunk 4]
```diff
@@ -19,6 +19,7 @@ class PICMI_Species(_ClassWithInit):
     Species
       - particle_type=None: A string specifying an elementary particle, atom, or other, as defined in the openPMD 2 species type extension, openPMD-standard/EXT_SpeciesType.md
       - name=None: Name of the species
+      - method: One of 'Boris', 'Vay', 'Higuera-Cary', 'Li' , 'free-streaming', and 'LLRK4' (Landau-Lifschitz radiation reaction formula with RK-4) (string)
```
Comment (dpgrote) [inline PICMI_Python/particles.py:22]: Following the pattern, this should be `method=None: One of ...`

**Review by dpgrote — COMMENTED** [review]
This is a good idea, adding the method argument for particle species. I have a few comments.

File: PICMI_Python/particles.py (hunk 4 above)
Comment (tsung1029) [reply PICMI_Python/particles.py]: Thank you!  I think I made all the changes according to the reviews.  Please give this version a look.

**Review by dpgrote — APPROVED** [review]
Great! Thanks!

**Review by dpgrote — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**